katas/content/teleportation/reconstruct_message_charlie/Solution.qs

```diff
@@ -0,0 +1,13 @@
+namespace Kata {
+    operation ReconstructMessageWhenThreeEntangledQubits(qCharlie : Qubit, (b1 : Bool, b2 : Bool), b3 : Bool) : Unit {
+        if b1 {
+            Z(qCharlie);
+        }
+        if b2 {
+            X(qCharlie);
+        }
+        if b3 {
+            X(qCharlie);
+        }
```

Comment:
We can actually simplify that, since two X gates will cancel each other out
```suggestion
        if b2 != b3 {
            X(qCharlie);
        }
```